helix-term/src/ui/editor.rs

```diff
@@ -654,6 +654,19 @@ impl EditorView {
     }
 }
 
+fn normalize_position(text: &Rope, row: usize, column: usize) -> usize {
+    let line_number = std::cmp::min(row, text.len_lines() - 2);
+    let line_start = text.line_to_char(line_number);
+    let line = text.line(line_number);
+    let mut col = 0;
+    if line_number == row && column >= OFFSET as usize {
+        col = std::cmp::min(column - OFFSET as usize, line.len_chars() - 1);
+    } else if line_number < row {
+        col = line.len_chars() - 1;
+    }
+    line_start + col
+}
+
```

Comment:
This doesn't really account for split views, graphemes or tabs.

You should use the raw coordinates and look them up on the view tree (for now you can probably just loop over `editor.tree.views()` and find the one with an overlapping `view.area`). Then there should be a complement method to https://github.com/helix-editor/helix/blob/a6aad3122d49ba07896ce047b5ae23da34eae803/helix-view/src/view.rs#L130-L162

`pos_at_screen_coords` that will give you the character position.

Reply:
Finished, thank you. Will add dragging on line numbers right now.